src/KOKKOS/fix_wall_lj93_kokkos.cpp

```diff
@@ -29,11 +40,156 @@ FixWallLJ93Kokkos<DeviceType>::FixWallLJ93Kokkos(LAMMPS *lmp, int narg, char **a
   kokkosable = 1;
   atomKK = (AtomKokkos *) atom;
   execution_space = ExecutionSpaceFromDevice<DeviceType>::space;
-  datamask_read = EMPTY_MASK;
-  datamask_modify = EMPTY_MASK;
-  virial_global_flag = virial_peratom_flag = 0;
+  datamask_read = X_MASK | V_MASK | MASK_MASK;
+  datamask_modify = F_MASK;
+
+  memoryKK->create_kokkos(k_epsilon,6,"wall_lj93:epsilon");
+  memoryKK->create_kokkos(k_sigma,6,"wall_lj93:sigma");
+  memoryKK->create_kokkos(k_cutoff,6,"wall_lj93:cutoff");
+
+  d_epsilon = k_epsilon.template view<DeviceType>();
+  d_sigma = k_sigma.template view<DeviceType>();
+  d_cutoff = k_cutoff.template view<DeviceType>();
+
+  for( int i=0 ; i<6 ; i++ ) {
+    k_epsilon.h_view(i) = epsilon[i];
+    k_sigma.h_view(i) = sigma[i];
+    k_cutoff.h_view(i) = cutoff[i];
+  }
+
+  k_epsilon.template modify<LMPHostType>();
+  k_sigma.template modify<LMPHostType>();
+  k_cutoff.template modify<LMPHostType>();
+
+  k_epsilon.template sync<DeviceType>();
+  k_sigma.template sync<DeviceType>();
+  k_cutoff.template sync<DeviceType>();
+
+  memoryKK->create_kokkos(d_coeff1,6,"wall_lj93:coeff1");
+  memoryKK->create_kokkos(d_coeff2,6,"wall_lj93:coeff2");
+  memoryKK->create_kokkos(d_coeff3,6,"wall_lj93:coeff3");
+  memoryKK->create_kokkos(d_coeff4,6,"wall_lj93:coeff4");
+  memoryKK->create_kokkos(d_offset,6,"wall_lj93:offset");
+
+  memoryKK->create_kokkos(k_ewall,ewall,7,"wall_lj93:ewall");
+  d_ewall = k_ewall.template view<DeviceType>();
+
+}
+
+template<class DeviceType>
+FixWallLJ93Kokkos<DeviceType>::~FixWallLJ93Kokkos()
+{
+  if (copymode) return;
+
+  memoryKK->destroy_kokkos(k_epsilon);
+  memoryKK->destroy_kokkos(k_sigma);
+  memoryKK->destroy_kokkos(k_cutoff);
+
+  memoryKK->destroy_kokkos(d_coeff1);
+  memoryKK->destroy_kokkos(d_coeff2);
+  memoryKK->destroy_kokkos(d_coeff3);
+  memoryKK->destroy_kokkos(d_coeff4);
+  memoryKK->destroy_kokkos(d_offset);
+
+  memoryKK->destroy_kokkos(k_vatom,vatom);
+  memoryKK->destroy_kokkos(k_ewall,ewall);
+
+}
+
+/* ---------------------------------------------------------------------- */
+
+template <class DeviceType>
+void FixWallLJ93Kokkos<DeviceType>::precompute(int m)
+{
+  d_coeff1[m] = 6.0 / 5.0 * d_epsilon[m] * powint(d_sigma[m], 9);
```

Comment:
Will crash on GPU since it is host CPU code referencing GPU memory.

Reply:
fixed by commit https://github.com/lammps/lammps/pull/4190/commits/cac6b5dc02afcff7b42c96a90cf829a4da537a2a